lalrpop/src/api/mod.rs

```diff
@@ -208,21 +212,24 @@ impl Configuration {
     /// Process all files in the current directory, which -- unless you
     /// have changed it -- is typically the root of the crate being compiled.
     pub fn process_current_dir(&self) -> Result<(), Box<dyn Error>> {
+        // If we can get a current dir, check to make sure session.in_dir either *wasn't* set, or
+        // wasn't set to that dir.  If we can't get a current dir, we'll error out in a moment
+        // anyways, and that's the bigger problem.
+        if let Ok(current_dir) = current_dir() {
+            if self.session.in_dir.is_some() && self.session.in_dir != Some(current_dir) {
+                eprintln!("Error: \"process_current_dir()\" contradicts previously set in_dir");
+                return Err(Box::new(io::Error::from(io::ErrorKind::InvalidInput)));
```

Comment:
What would be the difference between using `Error::from(..)` versus `Error::new(io::ErrorKind::InvalidInput, "The above error message here")`. I think the latter means that the our error message will be more apparent? My intuition could be mistaken

Reply:
I would think that if the user is using `.unwrap()` on the result, which I think is the recommended and normal approach, then the code as written would display the message and then panic, whereas if we change to your suggestion, it would panic with that message.  If the user is doing something else, then not displaying the message would allow them to suppress it.  Which I guess is desirable.  I'll update to that approach.